staging/src/k8s.io/client-go/util/jsonpath/parser.go

```diff
@@ -392,7 +394,7 @@ Loop:
 		if err != nil {
 			return err
 		}
-		cur.append(newFilter(leftParser.Root, rightParser.Root, value[2]))
+		cur.append(newFilter(leftParser.Root, rightParser.Root, value[2], valueText))
```

Comment:
I am not clear about why these changes to preserve valueText are necessary? I see it being used for something in the test diff.

Edit: I see now they are used to decompose the parsed texted to then recompose it to again valid jsonpath to parse again. I do not think that is a good idea.

Reply:
The reason I added these new value properties is because it seemed like a better idea to create some convenience properties upstream rather than implement logic downstream in kubectl (or wherever) to reconstruct the value from the other Node properties. This probably seems weird because normally you would expect the jsonpath query to be valid and return what you're looking for, but unfortunately we are not guaranteed that a property in a config file will be set already, and in fact it is important that we're able to work on a completely empty config file meaning nothing is set, for us to be able to replicate the functionality of the old syntax. The only way I could think of to accomplish this was to parse everything out into a node list and just rebuild until we reach the end of the list basically, setting things as we go.

Reply:
Would it be possible to add an option to jsonpath to create missing intermediate objects as it walks?

Reply:
I don't think that would be feasible. As you can see in this PR the logic required to accomplish that is complicated and highly specialized to the specific data structure that jsonpath is being used to query.

Reply:
> Would it be possible to add an option to jsonpath to create missing intermediate objects as it walks? Kinda like how allowMissingKeys was added as an option

I'm not sure that's desirable here, given that the primary use-case for JSONPath here is printing only. The proposed approach in this PR seems the most minimal, although I'm still thinking if this is correct approach. Only now I see the limitation of our jsonpath library :disappointed: